pkg/server/ca/manager.go

```diff
@@ -983,7 +983,12 @@ func GenerateServerCACSR(signer crypto.Signer, trustDomain spiffeid.TrustDomain,
 }
 
 func SelfSignX509CA(ctx context.Context, signer crypto.Signer, trustDomain spiffeid.TrustDomain, subject pkix.Name, notBefore, notAfter time.Time) (*X509CA, []*x509.Certificate, error) {
-	template, err := CreateServerCATemplate(trustDomain.ID(), signer.Public(), trustDomain, notBefore, notAfter, big.NewInt(0), subject)
+	serialNumber, err := x509util.NewSerialNumber()
```

Comment:
nit: may we add a validation on https://github.com/spiffe/spire/blob/main/pkg/server/ca/manager_test.go#L141 validating that serial number is not "0"?